packages/grid/_modules_/grid/components/toolbar/DensitySelector.tsx

```diff
@@ -25,17 +25,17 @@ export function DensitySelector() {
   const DensityOptions: Array<DensityOption> = [
     {
       icon: <DensityCompactIcon />,
-      label: 'Compact',
+      label: apiRef?.current.getText('densityOptionCompact') as string,
```

Comment:
I don't think that we can accept a component to **ever** render without its label, so `!` instead of `?` so it crashes if it ever happens. If it happens, I strongly think that we need to change the architecture, e.g. move to using the context or else.

```suggestion
      label: apiRef!.current.getText('densityOptionCompact') as string,
```

Reply:
I blame autocomplete.